parsing/dialects/dialect1/parserImpls.ftl

```diff
@@ -2382,3 +2382,164 @@ SqlCreateJoinIndex SqlCreateJoinIndex() :
             select, indices);
     }
 }
+
+SqlCall CaseN() :
+{
+    SqlNodeList nodes = new SqlNodeList(getPos());
+    SqlNode e;
+    NoCaseUnknown extraPartitionOption = null;
+}
+{
+    <CASE_N>
+    <LPAREN>
+    e = Expression(ExprContext.ACCEPT_SUB_QUERY)
+    {
+        nodes.add(e);
+    }
+    (
+        <COMMA>
+        e = Expression(ExprContext.ACCEPT_SUB_QUERY)
+        {
+            nodes.add(e);
+        }
+    )*
+    [
+        <COMMA>
+        (
+            LOOKAHEAD(3)
+            <NO> <CASE> <OR> <UNKNOWN>
+            { extraPartitionOption = NoCaseUnknown.NO_CASE_OR_UNKNOWN; }
+        |
+            LOOKAHEAD(3)
+            <NO> <CASE> <COMMA> <UNKNOWN>
+            { extraPartitionOption = NoCaseUnknown.NO_CASE_COMMA_UNKNOWN; }
+        |
+            <NO> <CASE> { extraPartitionOption = NoCaseUnknown.NO_CASE; }
+        |
+            <UNKNOWN> { extraPartitionOption = NoCaseUnknown.UNKNOWN; }
+        )
+    ]
+    <RPAREN>
+    {
+        return new SqlCaseN(getPos(), nodes, extraPartitionOption);
+    }
+}
+
+SqlCall RangeN() :
+{
+    SqlNode testExpression;
```

Comment:
nit: Can this be marked as final? (same for any other variables defined in this function)

Reply:
Fixed in https://github.com/googleinterns/calcite/pull/212/commits/2ac5c3010b51f82c19624c1a7aadff8015134548